source/slang/slang-ir-specialize.cpp

```diff
@@ -2439,6 +2439,13 @@ IRInst* specializeGenericImpl(
                 // value.
                 cloneInstDecorationsAndChildren(
                     &env, module, specializeInst, specializedVal);
+
+                // If specializedVal is a function, optimize it.
```

Comment:
```suggestion
                // Perform IR simplifications to fold constants in this specialized value if it is a function, so 
                // further specializations from the specialized function will have as simple specialization 
                // arguments as possible to avoid creating specializations that eventually simplified into 
                // the same thing.
```